clients/drcachesim/tests/drcachesim_unit_tests.cpp

```diff
@@ -313,6 +336,129 @@ LLC {
+  parent          memory
+}
+)MYCONFIG";
+    std::istringstream config_in(config);
+    cache_simulator_t cache_sim(&config_in);
+
+    // The cache config specified coherence, and the only level with
+    // multiple caches is L1.  So there should be 2 snooped caches.
+    TEST_EQ(cache_sim.get_num_snooped_caches(), 2);
+
+    // L1s are 1-way, while L2 and LLC are both 4-way.
+    // If we cycle through 4 conflicting lines multiple times, the L2 will
+    // hold all four lines and never evict anything to LLC: we expect all
+    // misses in L1, 4 misses and many hits in L2, 4 misses and no hits
+    // in LLC.
+
+    // Test 4 conflicting lines.
+    const int NUM_LOOPS = 16;
+    const int L2_ASSOC = 4;
+    const int LLC_ASSOC = 4;
+    const int LLC_SIZE = 64 * 1024;
+    const int ADDR_STRIDE = LLC_SIZE; // Maximize conflicts.
+    const int CONFLICTING_ADDRESSES = 4;
+    for (int n = 0; n < NUM_LOOPS; n++) {
+        for (int i = 0; i < CONFLICTING_ADDRESSES; ++i) {
+            if (!cache_sim.process_memref(make_memref(ADDR_STRIDE * i))) {
+                std::cerr << "drcachesim failed: " << cache_sim.get_error_string()
+                          << "\n";
+                exit(1);
+            }
+        }
+    }
+
+    // Define stats helper functions that are specific to this test config.
+    auto get_l2_metric = [&](metric_name_t metric) {
+        return cache_sim.get_cache_metric(metric, /*level=*/2, /*core=*/0,
+                                          cache_split_t::DATA);
+    };
+    auto get_llc_metric = [&](metric_name_t metric) {
+        return cache_sim.get_cache_metric(metric, /*level=*/3, /*core=*/0,
+                                          cache_split_t::DATA);
+    };
+
+    int l2_misses = get_l2_metric(metric_name_t::MISSES);
+    int l2_hits = get_l2_metric(metric_name_t::HITS);
+    int llc_misses = get_llc_metric(metric_name_t::MISSES);
+    int llc_hits = get_llc_metric(metric_name_t::HITS);
+
+    TEST_EQ(l2_misses, CONFLICTING_ADDRESSES);
+    TEST_EQ(l2_hits, (NUM_LOOPS - 1) * CONFLICTING_ADDRESSES);
+    TEST_EQ(llc_misses, l2_misses);
+    TEST_EQ(llc_hits, 0);
+
+    // Increasing to 8 conflicting lines means neither L2 nor LLC can hold all
+    // of the lines, but as a victim cache the LLC is additive and should hold
+    // L2's conflict evictions:  we expect 4 hits (from prior test) and the rest
+    // misses in L2, but 4 (new) misses and the rest hits in LLC.
+    const int MORE_CONFLICTING_ADDRESSES = 8;
+    for (int n = 0; n < NUM_LOOPS; n++) {
+        for (int i = 0; i < MORE_CONFLICTING_ADDRESSES; ++i) {
+            if (!cache_sim.process_memref(make_memref(ADDR_STRIDE * i))) {
+                std::cerr << "drcachesim unit_test_child_hits failed: "
+                          << cache_sim.get_error_string() << "\n";
+                exit(1);
+            }
+        }
+    }
+
+    std::cerr << "Results from second exclusive-cache sequence.\n";
+    cache_sim.print_results();
```

Comment:
Is this supposed to be only for verbose/diagnostic runs?  Or was it supposed to be printed after the first sequence too?

Reply:
Oops, yes, it was leftover from my development/debug.  Originally I printed both.  It's gone.